openedx/core/lib/token_utils.py

```diff
@@ -1,124 +1,31 @@
-"""Utilities for working with ID tokens."""
-import json
-from time import time
-
-from django.conf import settings
-from django.utils.functional import cached_property
-from jwkest import jwk
-from jwkest.jws import JWS
-
-from edx_django_utils.monitoring import set_custom_metric
-from student.models import UserProfile, anonymous_id_for_user
+"""
+Deprecated JwtBuilder class for backward compatibility until edx-enterprise is updated.
+"""
+from openedx.core.djangoapps.oauth_dispatch.jwt import create_api_client_jwt
 
 
 class JwtBuilder(object):
-    """Utility for building JWTs.
-
-    Unifies diverse approaches to JWT creation in a single class. This utility defaults to using the system's
-    JWT configuration.
-
-    NOTE: This utility class will allow you to override the signing key and audience claim to support those
-    clients which still require this. This approach to JWT creation is DEPRECATED. Avoid doing this for new clients.
-
-    Arguments:
-        user (User): User for which to generate the JWT.
-
-    Keyword Arguments:
-        asymmetric (Boolean): Whether the JWT should be signed with this app's private key.
-        secret (string): Overrides configured JWT secret (signing) key. Unused if an asymmetric signature is requested.
-        issuer (string): Overrides configured JWT issuer.
     """
-
-    def __init__(self, user, asymmetric=False, secret=None, issuer=None):
+    Deprecated. Use openedx.core.djangoapps.oauth_dispatch.jwt.JwtBuilder directly.
+    """
+    def __init__(self, user, secret=None):
```

Comment:
Are these other arguments not used by edx-enterprise?  Is that why it is safe to change the signature on something being left for backward compatibility?

Reply:
yes.